sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/JsonUtils.java

```diff
@@ -73,6 +78,77 @@ public String apply(Row input) {
     };
   }
 
+  public static Schema beamSchemaFromJsonSchema(String jsonSchemaStr) {
+    org.everit.json.schema.ObjectSchema jsonSchema = jsonSchemaFromString(jsonSchemaStr);
+    return beamSchemaFromJsonSchema(jsonSchema);
+  }
+
+  private static Schema beamSchemaFromJsonSchema(org.everit.json.schema.ObjectSchema jsonSchema) {
+    Schema.Builder beamSchemaBuilder = Schema.builder();
+    for (String propertyName : jsonSchema.getPropertySchemas().keySet()) {
+      org.everit.json.schema.Schema propertySchema =
+          jsonSchema.getPropertySchemas().get(propertyName);
+      if (propertySchema == null) {
+        throw new IllegalArgumentException("Unable to parse schema " + jsonSchema.toString());
+      }
+      if (propertySchema.getClass().equals(org.everit.json.schema.ObjectSchema.class)) {
+        beamSchemaBuilder =
+            beamSchemaBuilder.addField(
+                Schema.Field.of(propertyName, beamTypeFromJsonSchemaType(propertySchema)));
+      } else if (propertySchema.getClass().equals(org.everit.json.schema.ArraySchema.class)) {
+        beamSchemaBuilder =
+            beamSchemaBuilder.addField(
+                Schema.Field.of(
+                    propertyName,
+                    Schema.FieldType.array(
+                        beamTypeFromJsonSchemaType(
+                            ((ArraySchema) propertySchema).getAllItemSchema()))));
```

Comment:
hmmm good observation - these don't seem to translate well to Beam schemas. since the individual 'fields' in a tuple don't have a name, they can't be a Beam Row.

I've just added validation to not support them and validation for the error message.